rest/route/reliability_test.go

```diff
@@ -402,3 +448,165 @@ func TestRunTestHandler(t *testing.T) {
 	lastDoc := sc.MockTaskReliabilityConnector.CachedTaskReliability[100]
 	assert.Equal(lastDoc.StartAtKey(), resp.Pages().Next.Key)
 }
+
+func setupEnv(ctx context.Context) (*mock.Environment, error) {
+	env := &mock.Environment{}
+
+	if err := env.Configure(ctx, "", nil); err != nil {
+		return nil, errors.WithStack(err)
+	}
+	return env, nil
+}
+
+func withSetupAndTeardown(t *testing.T, env evergreen.Environment, fn func()) {
+	require.NoError(t, db.ClearCollections(stats.DailyTaskStatsCollection))
+	defer func() {
+		assert.NoError(t, db.ClearCollections(stats.DailyTaskStatsCollection))
+	}()
+
+	fn()
+}
+
+func TestReliability(t *testing.T) {
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for opName, opTests := range map[string]func(ctx context.Context, t *testing.T, env evergreen.Environment){
+		"Pagination": func(ctx context.Context, t *testing.T, env evergreen.Environment) {
+			pageSize := 50
+			for testName, testCase := range map[string]func(ctx context.Context, t *testing.T){
+				"Less Than One Page": func(ctx context.Context, t *testing.T) {
+					assert := assert.New(t)
+					err := setupTest(t)
+					assert.NoError(err)
+					url := getURL(projectID, map[string]interface{}{
+						"tasks":         "aggregation_expression_multiversion_fuzzer",
+						"after_date":    "2019-01-02",
+						"group_by_days": "10",
+					})
+
+					sc := &data.MockConnector{
+						MockTaskReliabilityConnector: data.MockTaskReliabilityConnector{},
+						URL:                          url,
+					}
+					handler := makeGetProjectTaskReliability(sc).(*taskReliabilityHandler)
+					assert.NoError(err)
+
+					// 1 page size of documents are available but 2 page sizes requested.
+					sc.MockTaskReliabilityConnector.SetTaskReliabilityScores("aggregation_expression_multiversion_fuzzer", pageSize)
+					handler.filter = reliability.TaskReliabilityFilter{
+						StatsFilter: stats.StatsFilter{
+							Limit: pageSize * 2,
+						},
+					}
+
+					resp := handler.Run(context.Background())
```

Comment:
use the context passed in rather than making a new one.

Reply:
:+1: